libs/network-stats/src/components/stats-manager/stats-manager.tsx

```diff
@@ -14,16 +14,81 @@ interface statsManagerProps {
   className?: string;
 }
 
+interface dataSourceProps {
+  title: string;
+  stats: string;
+  nodes: string;
+}
+
+const dataSources = {
+  mainnet: {
+    title: 'Mainnet',
+    stats: 'https://api.token.vega.xyz/statistics',
+    nodes: 'https://api.token.vega.xyz/nodes-data',
+  },
+  testnet: {
+    title: 'Testnet',
+    stats: 'https://lb.testnet.vega.xyz/statistics',
+    nodes: 'https://lb.testnet.vega.xyz/datanode/rest/nodes-data',
+  },
+  devnet: {
+    title: 'Devnet',
+    stats: 'https://n04.d.vega.xyz/datanode/rest/statistics',
+    nodes: 'https://n04.d.vega.xyz/datanode/rest/nodes-data',
+  },
+  stagnet1: {
+    title: 'Stagnet 1',
+    stats: 'https://n03.s.vega.xyz/datanode/rest/statistics',
+    nodes: 'https://n03.s.vega.xyz/datanode/rest/nodes-data',
+  },
+  stagnet2: {
+    title: 'Stagnet 2',
+    stats: 'https://n01.stagnet2.vega.xyz/datanode/rest/statistics',
+    nodes: 'https://n01.stagnet2.vega.xyz/datanode/rest/nodes-data',
+  },
+};
```

Comment:
chore (blocking): can't hard code this. We can add environments or remove them at any point, we don't want to have to deploy the frontend if an evironment is changed. You should use environment variables here, one fo the REST url (bear in mind for mainnet this could be anything) and a network (which also could be anything).

The API paths are consistent so you need:
```
const title = process.env['NX_VEGA_ENV']
const endpoint = process.env['NX_VEGA_REST']
return {
  title,
  stats: new URL(endpoint, '/statistics').href
  nodes: new URL(endpoint, '/nodes-data').href
}
```

Then adjust the different environment files

Reply:
Also this component shouldn't care about vega environments ect. hoist this logic to the consumer, take these variables as props to the components I think makes the most sense.

Reply:
Great points, thanks